src/core-sdk-tasks/Crossgen.cs

```diff
@@ -133,33 +131,23 @@ protected override string GenerateFullPathToTool()
                 return CrossgenPath;
             }
 
-            return "crossgen";
+            return "crossgen2";
         }
 
         protected override string GenerateCommandLineCommands()
         {
-            if (_secondInvocationToCreateSymbols)
-            {
-                return $"{GetReadyToRun()} {GetPlatformAssemblyPaths()} {GetDiasymReaderPath()} {GetCreateSymbols()}";
-            }
-
-            return $"{GetReadyToRun()} {GetMissingDependenciesOk()} {GetInPath()} {GetOutPath()} {GetPlatformAssemblyPaths()} {GetJitPath()}";
+            return $"{GetInPath()} {GetOutPath()} {GetArchitecture()} {GetPlatformAssemblyPaths()} {GetCreateSymbols()}";
```

Comment:
@AntonLapounov @trylek I just realized we are not specifying `-O` option to optimize generate code. does cg2 now look at the PE info and optimize unless its compiled with `Debug`? If not we should explicitly specify that.

Reply:
Yes, it does: https://github.com/dotnet/runtime/blob/17481fef502ee6aed6f9e8fc76e45bb5863c68b5/src/coreclr/tools/aot/crossgen2/Program.cs#L648

Reply:
Ok nice, just wanted to make sure that was indeed implemented.

Reply:
Well, optimization detection is indeed implemented in CG2 but according to Anton's link it only kicks in for single-input builds, if the same code is used for composite compilation, that probably still needs the explicit -O option.

Reply:
We do not use composite mode here.